iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/unique_id.hpp

```diff
@@ -45,7 +45,10 @@ class UniqueId : public cxx::NewType<uint64_t,
     UniqueId() noexcept;
 
   private:
-    static std::atomic<value_type> g_IdCounter; // initialized in corresponding cpp file
+    /// @NOLINTJUSTIFICATION only accessible by this class. the global variable is required to
```

Comment:
AFAIK the `@` can be removed. I talked about this with @elBoberido recently. Same for all other suppressions.

Reply:
@elfenpiff Oh and could you maybe document the suppression syntax for clang tidy in `CONTRIBUTING.md`?

Reply:
I added it